OpenSim/Common/ArrayPtrs.h

```diff
@@ -1024,6 +1064,121 @@ int searchBinary(const T &aObject,bool aFindFirst=false,
 //=============================================================================
 };  // END of class ArrayPtrs
 
+
+//=============================================================================
+// Custom Iterator
+//=============================================================================
+//_____________________________________________________________________________
+
+/**
+* Inner class that describes a const_iterator and 'regular' iterator at the same time,
+* depending on the boolean template parameter (default: true - a const_iterator)
+*
+* With some modifications from:
+* http://www.sj-vs.net/c-implementing-const_iterator-and-non-const-iterator-without-code-duplication/
+*
+* @author Dimitar Stanev
+*/
+template<class T, bool is_const_iterator = true>
+class ArrayPtrsIterator : public std::iterator<std::forward_iterator_tag, T>
+{
+    /**
+    * For const_iterator:   define ArrayPointerType to be a   pointer to a const pointer to T
+    * For regular iterator: define ArrayPointerType to be a   T**
+    */
+    typedef typename std::conditional<is_const_iterator, T* const *, T**>::type ArrayPointerType;
```

Comment:
Also, I am not very confident with T* const *, as I accidentally find a way to bypass the compiler errors.